app/models/custom_field/order_statements.rb

```diff
@@ -167,4 +169,9 @@ def join_for_order_by_version_sql
       multi_value:
     )
   end
+
+  def join_for_order_by_hierarchy_sql
+    table_name = CustomField::Hierarchy::Item.quoted_table_name
+    join_for_order_sql(value: "item.label", join: "INNER JOIN #{table_name} item ON item.id = cv.value::bigint")
```

Comment:
⚠️ Do I read this correctly, that this is now sorted by label (as it is joined into the values table). So we should keep that in mind, whenever we decide to change the display representation in the wp table view. (e.g. no longer starting with the label, but some shorts of ancestors).

Reply:
I just implemented the simples possible way forward.

To be fair this could also be extended to handle multi values... I think it's fair to bring to discussion.